core/Context.cc

```diff
@@ -111,34 +111,28 @@ GlobalSubstitution::GlobalSubstitution(const GlobalState &from, GlobalState &to,
     }
 
     if (!fastPath || debug_mode) {
-        bool seenEmpty = false;
```

Comment:
I feel like we should have been able to catch this weird dead code sooner.

Reply:
Ah this isn't dead code. This handles the special 0-indexed case which I have removed!

Reply:
Ah yep -- I misread. Thanks!